R/outputs.R

```diff
@@ -1276,3 +1276,25 @@ get_period_metadata <- function(calendar_quarters) {
       quarter_label = quarter_year_labels(quarter_id)
     )
 }
+
+
+#' Read hintr output
+#'
+#' Read the hintr model output or plot data saved as a qs or an rds file.
+#' This is the data saved from hintr_run_model or hintr_calibrate
+#' before the output zip is generated. This uses the file extension
+#' to identify the reading function to use.
+#'
+#' @param path Path to the file
+#'
+#' @return The read data
+#' @export
+read_hintr_output <- function(path) {
+  is_qs <- grepl("\\.qs$", path, ignore.case = TRUE)
```

Comment:
there is a builtin function for this: `tools::file_ext()`
```
> tools::file_ext
function (x) 
{
    pos <- regexpr("\\.([[:alnum:]]+)$", x)
    ifelse(pos > -1L, substring(x, pos + 1L), "")
}
```

Not sure there's any benefit to preferring that to `grepl()` though.